src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -0,0 +1,319 @@
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+
+    String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+    GetInspectTemplateRequest request = GetInspectTemplateRequest.newBuilder().setName(templateName).build();
+
+    try {
+      InspectTemplate template = client.getInspectTemplate(request);
+
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+        "Unable to validate template name. Ensure template ID matches the specified ID in DLP");
+    }
+  }
+
+  /**
+   * Splitter Transform splits the sensitive and non-sensitive record into different ports. If user has selected entire
+   * record to be checked for sensitive data, then all the fields are concacted as string and passed  to data loss
+   * prevention API.
+   *
+   * @param record a <code>StructuredRecord</code> being passed from the previous stage.
+   * @param emitter a <code>MultiOutputEmitter</code> to emit sensitive or non-sensitive data on different ports.
+   */
+  @Override
+  public void transform(StructuredRecord record, MultiOutputEmitter<StructuredRecord> emitter) throws Exception {
+    String recordString = null;
+    ContentItem contentItem = null;
+
+    if (!config.isEntireRecord()) {
+      recordString = record.get(config.getFieldName()).toString();
+    } else {
+      recordString = StructuredRecordStringConverter.toDelimitedString(record, ",");
+    }
+    try {
+
+      // depending on input schema field object
+      contentItem = ContentItem.newBuilder().setValue(recordString).build();
+
+      String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+
+      InspectContentRequest request =
+        InspectContentRequest.newBuilder()
+                             .setParent(ProjectName.of(config.getProject()).toString())
+                             .setInspectTemplateName(templateName)
+                             .setItem(contentItem)
+                             .build();
+
+      metrics.count("dlp.requests.count", 1);
+      InspectContentResponse response = client.inspectContent(request);
+      InspectResult result = response.getResult();
+
+      metrics.count("dlp.requests.success", 1);
+      if (result.getFindingsList().size() > 0) {
+        emitter.emit(SENSITIVE_PORT, record);
+        return;
+      }
+
+      emitter.emit(NON_SENSITIVE_PORT, record);
+
+    } catch (Exception e) {
+      metrics.count("dlp.requests.fail", 1);
+      if (e instanceof ResourceExhaustedException) {
+        LOG.error(
+          "Failed due to DLP rate limit, please request more quota from DLP: https://cloud.google"
+            + ".com/dlp/limits#increases");
+        throw e;
+      }
+
+      switch (config.onErrorHandling()) {
+        case -1:
+          throw new Exception("Terminating pipeline on error as set in plugin configuration." + e.getMessage());
```

Comment:
is this an error path? wouldnt this mean it will throw exception every time input is processed?

Reply:
This is inside the catch block of the try-catch so it would throw an error if they user chose the "stop pipeline" option in the plugin config AND this is not a rate-limiting error